test-case/check-audio-equalizer.sh

```diff
@@ -0,0 +1,144 @@
+#!/bin/bash
+
+##
+## Case Name: check-audio-equalizer.sh
+## Preconditions:
+##    SOF Topology should have EQ component(s) in any pipeline
+## Description:
+##    check IIR/FIR config working
+## Case step:
+##    1. Check if the topology has a EQ component in it
+##    2. Test with IIR config list
+##    3. Test with FIR config list
+## Expect result:
+##    sof-ctl and aplay should return 0
+##
+
+# source from the relative path of current folder
+source $(dirname ${BASH_SOURCE[0]})/../case-lib/lib.sh
+
+OPT_OPT_lst['t']='tplg'     OPT_DESC_lst['t']='tplg file, default value is env TPLG: $TPLG'
+OPT_PARM_lst['t']=1         OPT_VALUE_lst['t']="$TPLG"
+
+OPT_OPT_lst['d']='duration' OPT_DESC_lst['d']='aplay duration in second'
+OPT_PARM_lst['d']=1         OPT_VALUE_lst['d']=5
+
+OPT_OPT_lst['l']='loop'     OPT_DESC_lst['l']='loop count'
+OPT_PARM_lst['l']=1         OPT_VALUE_lst['l']=1
+
+OPT_OPT_lst['s']='sof-logger'   OPT_DESC_lst['s']="Open sof-logger trace the data will store at $LOG_ROOT"
+OPT_PARM_lst['s']=0             OPT_VALUE_lst['s']=1
+
+func_opt_parse_option $*
+tplg=${OPT_VALUE_lst['t']}
+duration=${OPT_VALUE_lst['d']}
+loop_cnt=${OPT_VALUE_lst['l']}
+
+func_pipeline_export $tplg "type:playback"
+sofcard=${SOFCARD:-0}
+
+# Test equalizer
+func_test_eq()
+{
+    local id=$1
+    local conf=$2
+
+    [ ! -f $conf ] && dloge "$conf not exist" && exit 1
+
+    dlogc "sof-ctl -Dhw:$sofcard -n $id -s $conf"
+    sof-ctl -Dhw:$sofcard -n $id -s $conf
+    if [ $? -ne 0 ]; then
+        dloge "Equalizer setting failure with $conf"
+        return 1
+    fi
+    # test with aplay
+    dlogc "aplay -D$dev -f $fmt -c $channel -r $rate /dev/zero -d $duration"
+    aplay -D$dev -f $fmt -c $channel /dev/zero -d $duration
+    if [ $? -ne 0 ]; then
+        dloge "Equalizer test failure with $conf"
+        return 1
+    fi
+    sleep 1
+}
+
+current_dir="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
```

Comment:
What is this? What you want to get here? Won't the following code be enough?
```suggestion
current_dir=$(dirname ${BASH_SOURCE[0]})
```